### ISSUE TITLE
No option to download a pivoted CSV from dashboard view for a pivot table chart

### ISSUE THREAD

**Issue by sharpsell-product** [apache/superset#26142]
When I edit the pivot table chart, I can see 2 options of CSV download on the top right corner - Original CSV and Pivoted CSV , but when I click on download from dashboard view for the same chart, it only shows option to download as a simple CSV file. How can I get the option to download as pivoted CSV from the dashboard itself?

How to reproduce the bug

1. Go to '.charts.'
2. Create a pivot table chart with both rows, columns, and metrics
3. add the chart to any dashboard with option to download the CSV of the chart from the dashboard itself
4. click on the download option

 
Expected results
1. When I click on the download option I should be able to find an option to download as a pivoted CSV

Actual results
1. I only see the option to download as an original CSV. This option downloads the data as a normal table with columns converted into rows

![image](https://github.com/apache/superset/assets/95677763/bd8576c7-4800-49de-8f66-eeb0e1ae0e82)

![image](https://github.com/apache/superset/assets/95677763/a36251c1-be67-4f2e-8ca9-89abf520ebae)

### Environment

About
Powered by Apache Superset
Version: 2.1.0
Build: 2.1.0

- browser type and version: Chrome ( Version 119.0.6045.160 (Official Build) (64-bit))
- superset version: `superset version` : Version: 2.1.0 , Build: 2.1.0
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active: Yes

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any. 
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by RavshanbekXB** [comment]
I also need this feature

**Comment by adimyth** [comment]
@rusackas any ideas on this one?

**Comment by rusackas** [comment]
Nope... it's come up on a few other Issues, so we should consolidate those. We might also consider moving this to a Discussion since it's more of a feature request than a bug report. PRs welcome, as always, as this would undoubtedly be a nice thing to have. Nut sure if @yousoph is aware of any progress on this front.
